### PR TITLE
🎤 Create notifications example

### PR DESCRIPTION
**Pull request by nezouse**

### CODE REVIEW

File: packages/example/src/components/Notifications/Forms.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+const wethInterface = new utils.Interface(abi)
+
+interface TransferFormProps {
+  balance: BigNumber | undefined
+  send: (value: BigNumber) => void
+  title: string
+  ticker: string
+}
+
+const TransferForm = ({ balance, send, title, ticker }: TransferFormProps) => {
```
Comment (marekkirejczyk) [inline packages/example/src/components/Notifications/Forms.tsx:27]: TransactionForm

File: packages/example/src/components/Notifications/History.tsx [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    <Table>{children}</Table>
+  </ContentBlock>
+)
+
+interface TransactionListProps {
+  chainId: ChainId
+}
+
+export const TransactionsList = ({ chainId }: TransactionListProps) => {
+  const { transactions } = useTransactionsContext()
```
Comment (marekkirejczyk) [inline packages/example/src/components/Notifications/History.tsx:25]: Why not just `useTransactions()`

File: packages/example/src/components/Notifications/History.tsx [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+  const chainTransactions = transactions[chainId] ?? []
+  const reversed = [...chainTransactions].reverse()
```
Comment (marekkirejczyk) [inline packages/example/src/components/Notifications/History.tsx:27]: Let's do reverse by default (modify reducer).

Idea to discuss:
```ts
useTransactions({chainId: ...})
```
**ChainId**
- chainId is by default null, in which case useTransaction will use useEthers to determine chainId.
- chainId can be single chainId 
- in the future: array of chainIds
- question: how go we show all chainIds? empty array? 0? other?



```ts
{chainId: null }
```

File: packages/example/src/components/Notifications/History.tsx [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+
+  return (
+    <TableWrapper title="Transactions history">
+      {reversed.map((tx) => (
+        <div key={tx.transaction.hash}>
+          {!tx.receipt && <Spinner />}
+          <span>{tx.transaction.hash.slice(0, 6) + '...' + tx.transaction.hash.slice(-4)}</span>
```
Comment (marekkirejczyk) [inline packages/example/src/components/Notifications/History.tsx:34]: Use [shortenAddress(...)](https://usedapp.readthedocs.io/en/latest/core.html#shortenaddress)

File: packages/example/src/components/Notifications/History.tsx [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+interface NotificationPanelProps {
+  title: string
+  icon: ReactNode
+  transactionHash: string
+}
+
+const NotificationPanel = ({ title, icon, transactionHash }: NotificationPanelProps) => {
+  function getEtherscanLink(hash: string) {
+    return `https://kovan.etherscan.io/tx/${hash}`
+  }
```
Comment (marekkirejczyk) [inline packages/example/src/components/Notifications/History.tsx:50]: Use [getExplorerAddressLink](https://usedapp.readthedocs.io/en/latest/core.html#getexploreraddresslink)

File: packages/example/src/components/Notifications/History.tsx [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+    </NotificationWrapper>
+  )
+}
+
+interface NotificationItemProps {
+  notification: Notification
+}
+
+const NotificationItem = ({ notification }: NotificationItemProps) => {
+  switch (notification.type) {
```
Comment (marekkirejczyk) [inline packages/example/src/components/Notifications/History.tsx:69]: You should be able to do a switch/mapping just on the icon, i.e.
```ts
const label = {'transactionStarted': "Transaction started", ....}
...

function NotificationIcon() {
  switch(type) {
    case 'transactionStarted': return (<ClockIcon />)
    ...
  }
}
```

File: packages/example/src/components/Notifications/History.tsx [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+    default:
+      return null
+  }
+}
+interface NotificationsListProps {
+  chainId: ChainId
+}
+
+export const NotificationsList = ({ chainId }: NotificationsListProps) => {
+  const { notifications } = useNotificationsContext()
```
Comment (marekkirejczyk) [inline packages/example/src/components/Notifications/History.tsx:103]: Why not just `useNotifications`?

File: packages/example/src/components/Notifications/History.tsx [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+  const chainNotifications = notifications[chainId] ?? []
+  const reversed = [...chainNotifications].reverse()
```
Comment (marekkirejczyk) [inline packages/example/src/components/Notifications/History.tsx:105]: Ascending order by default.

File: packages/example/src/components/TopBar.tsx [hunk 9]
```diff
@@ -27,6 +27,10 @@ export function TopBar() {
               {' '}
               Tokens{' '}
             </HeaderLink>
+            <HeaderLink activeClassName="active-page" to="/notifications">
```
Comment (marekkirejczyk) [inline packages/example/src/components/TopBar.tsx:30]: Let's call it Transactions instead of Notifications

**Review by marekkirejczyk — CHANGES_REQUESTED** [review]
Looks very good. See proposed changes in the comments.

File: packages/example/src/components/Notifications/History.tsx (hunk 6 above)
Comment (nezouse) [reply packages/example/src/components/Notifications/History.tsx:69]: I don't know if I understand this correctly, but anyway I need to check the type of transaction just to get hash from it.
The only thing that I can get from notification by default right now is `type`.

File: packages/example/src/components/Notifications/History.tsx (hunk 4 above)
Comment (marekkirejczyk) [reply packages/example/src/components/Notifications/History.tsx:34]: ahhh, shall we add shortenTransaction equvalent?

Comment (nezouse) [reply packages/example/src/components/Notifications/History.tsx:34]: We can just remove checking if the address is correct (and checksumming it). Anyway, it should just work on any string which is long enough.

File: packages/example/src/components/Notifications/History.tsx [hunk 10]
```diff
@@ -0,0 +1,201 @@
...
+    case 'transactionSucceed':
+      return (
+        <NotificationPanel
+          title="Transaction succeed"
+          icon={<CheckIcon />}
+          transactionHash={notification.transaction.hash}
+        />
+      )
+    default:
+      return null
```
Comment (marekkirejczyk) [inline packages/example/src/components/Notifications/History.tsx:105]: Next PR: Add `walletConnected`.

**Review by marekkirejczyk — APPROVED** [review]

**Pull request merged**